xarray/tests/test_backends.py

```diff
@@ -1073,6 +1075,79 @@ class NetCDF4ViaDaskDataTestAutocloseTrue(NetCDF4ViaDaskDataTest):
     autoclose = True
 
 
+@requires_zarr
+class BaseZarrTest(CFEncodedDataTest):
+    def test_auto_chunk(self):
+        original = create_test_data().chunk()
+
+        with self.roundtrip(
+                original, open_kwargs={'auto_chunk': False}) as actual:
+            for k, v in actual.variables.items():
+                # only index variables should be in memory
+                self.assertEqual(v._in_memory, k in actual.dims)
+                # there should be no chunks
+                self.assertEqual(v.chunks, None)
+
+        with self.roundtrip(
+                original, open_kwargs={'auto_chunk': True}) as actual:
+            for k, v in actual.variables.items():
+                # only index variables should be in memory
+                self.assertEqual(v._in_memory, k in actual.dims)
+                # chunk size should be the same as original
+                self.assertEqual(v.chunks, original[k].chunks)
+
+    def test_chunk_encoding(self):
+        data = create_test_data()
+        chunks = (5, 5)
+        data['var2'].encoding.update({'chunks': chunks})
+        with self.roundtrip(data) as actual:
+            self.assertEqual(chunks, actual['var2'].encoding['chunks'])
+        data['var2'].encoding.update({'chunks': (5, 4.5)})
+        with pytest.raises(ValueError):
+            with self.roundtrip(data) as actual:
+                pass
+
+    def test_vectorized_indexing(self):
+        self._test_vectorized_indexing(vindex_support=True)
+
+
+    def test_dataset_caching(self):
+        # TODO: someone who understand caching figure out whether chaching
+        # makes sense for Zarr backend
+        pass
```

Comment:
I would say yes, if Dask is not required for using zarr with xarray. But I'm happy to leave this to later.